src/java/org/apache/nutch/parse/ParseOutputFormat.java

```diff
@@ -379,15 +378,16 @@ public static String filterNormalize(String fromUrl, String toUrl,
       if (ignoreInternalLinks) {
         if ("bydomain".equalsIgnoreCase(ignoreExternalLinksMode)) {
           String toDomain = URLUtil.getDomainName(targetURL).toLowerCase();
-          //FIXME: toDomain will never be null, correct?
           if (toDomain == null || toDomain.equals(origin)) {
             return null; // skip it
           }
         } else {
           String toHost = targetURL.getHost().toLowerCase();
-          //FIXME: toDomain will never be null, correct?
           if (toHost == null || toHost.equals(origin)) {
-            return null; // skip it
+            if (exemptionFilters == null // check if it is exempted?
```

Comment:
The [exemption filter|https://builds.apache.org/job/nutch-trunk/javadoc/org/apache/nutch/urlfilter/ignoreexempt/package-summary.html] is to define exemption for otherwise skipped external links. Should not be in the branch which handles internal links, otherwise we would have to redefine the exemption filter interface.

Reply:
Thank you for this review @sebastian-nagel , I will be attacking this over this weekend. I appreciate the clarification in understanding. This project has absolutely fantastic reviewers, thank you again!